ee/vellum_ee/workflows/display/workflows/base_workflow_display.py

```diff
@@ -320,9 +321,32 @@ def serialize(self) -> JsonObject:
 
         edges.extend(synthetic_output_edges)
 
+        nodes_list = list(serialized_nodes.values())
+        nodes_dict_list = [cast(Dict[str, Any], node) for node in nodes_list if isinstance(node, dict)]
+
+        all_nodes_at_zero = all(
+            (
+                isinstance(node.get("display_data"), dict)
+                and isinstance(node["display_data"].get("position"), dict)
+                and node["display_data"]["position"].get("x", 0) == 0.0
+                and node["display_data"]["position"].get("y", 0) == 0.0
+            )
+            for node in nodes_dict_list
+        )
+
+        should_apply_auto_layout = (
+            all_nodes_at_zero
+            and len(nodes_dict_list) > 0
+            and len(edges) > 0
+            and getattr(self._workflow, "_enable_auto_layout", False)
+        )
+
+        if should_apply_auto_layout:
+            self._apply_auto_layout(nodes_dict_list, edges)
```

Comment:
Wrap this method in a try/except, adding the error to workflow context if there is one